server/chat-plugins/scavengers.ts

```diff
@@ -155,6 +155,7 @@ class PlayerLadder extends Ladder {
 	}
 
 	// add the different keys to the history - async for larger leaderboards
+	// FIXME: this is not what "async" means
 	softReset() {
```

Comment:
@sparkychildcharlie Making something async does not prevent it from causing freezes, or do anything at all, by itself.

If you split it into multiple Promises, though, it could help. But since this is one big Promise, it's actually no difference. Promises only change the order things happen in, they don't add multitasking by themselves.

The part that's confusing is that there are a lot of built-in functions, like everything in `fs`, which do add multitasking to Promises. But those have multi-tasking implemented inside them, it's not _just_ the Promise that makes it multitask. In PS, we have a library (process-manager) that will add multitasking to a Promise. But just creating a Promise is not enough.

I don't know if I'm explaining this well. This YouTube video might help: https://www.youtube.com/watch?v=8aGhZQkoFbQ